cosmic-applet-minimize/src/wayland_handler.rs

```diff
@@ -0,0 +1,554 @@
+        screencopy_session.commit(zcosmic_screencopy_session_v1::Options::empty());
+        self.conn.flush().unwrap();
+
+        // TODO: wait for server to release buffer?
+        let res = session
+            .wait_while(|data| data.res.is_none())
+            .res
+            .take()
+            .unwrap();
+        pool.destroy();
+        buffer.destroy();
+
+        //std::thread::sleep(std::time::Duration::from_millis(16));
+
+        if res.is_ok() {
+            Some(ShmImage {
+                fd,
+                width: buffer_info.width,
+                height: buffer_info.height,
+            })
+        } else {
+            None
+        }
+    }
+}
+
+pub struct ShmImage<T: AsFd> {
+    fd: T,
+    pub width: u32,
+    pub height: u32,
+}
+
+impl<T: AsFd> ShmImage<T> {
+    pub fn image(&self) -> anyhow::Result<image::RgbaImage> {
+        let mmap = unsafe { memmap2::Mmap::map(&self.fd.as_fd())? };
+        image::RgbaImage::from_raw(self.width, self.height, mmap.to_vec())
+            .ok_or_else(|| anyhow::anyhow!("ShmImage had incorrect size"))
+    }
+}
+
+impl ProvidesRegistryState for AppData {
+    fn registry(&mut self) -> &mut RegistryState {
+        &mut self.registry_state
+    }
+
+    sctk::registry_handlers!();
+}
+
+impl SeatHandler for AppData {
+    fn seat_state(&mut self) -> &mut sctk::seat::SeatState {
+        &mut self.seat_state
+    }
+
+    fn new_seat(&mut self, _: &Connection, _: &QueueHandle<Self>, _: WlSeat) {}
+
+    fn new_capability(
+        &mut self,
+        _: &Connection,
+        _: &QueueHandle<Self>,
+        _: WlSeat,
+        _: sctk::seat::Capability,
+    ) {
+    }
+
+    fn remove_capability(
+        &mut self,
+        _: &Connection,
+        _: &QueueHandle<Self>,
+        _: WlSeat,
+        _: sctk::seat::Capability,
+    ) {
+    }
+
+    fn remove_seat(&mut self, _: &Connection, _: &QueueHandle<Self>, _: WlSeat) {}
+}
+
+impl ToplevelManagerHandler for AppData {
+    fn toplevel_manager_state(&mut self) -> &mut cctk::toplevel_management::ToplevelManagerState {
+        &mut self.toplevel_manager_state
+    }
+
+    fn capabilities(
+        &mut self,
+        _: &Connection,
+        _: &QueueHandle<Self>,
+        _: Vec<WEnum<zcosmic_toplevel_manager_v1::ZcosmicToplelevelManagementCapabilitiesV1>>,
+    ) {
+        // TODO capabilities could affect the options in the applet
+    }
+}
+impl AppData {
+    fn send_image(&self, handle: ZcosmicToplevelHandleV1) {
+        let tx = self.tx.clone();
+        let capure_data = CaptureData {
+            qh: self.queue_handle.clone(),
+            conn: self.conn.clone(),
+            wl_shm: self.shm_state.wl_shm().clone(),
+            screencopy_manager: self.screencopy_state.screencopy_manager.clone(),
+        };
+        std::thread::spawn(move || {
+            use std::ffi::CStr;
+            let name = unsafe { CStr::from_bytes_with_nul_unchecked(b"pipewire-screencopy\0") };
```

Comment:
```suggestion
            let name = unsafe { CStr::from_bytes_with_nul_unchecked(b"minimize-applet-screencopy\0") };
```

Just for easier debugging.